src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/permission-tab/components/form/main.js

```diff
@@ -0,0 +1,175 @@
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+define(['config', 'sulusecurity/collections/roles'], function(Config, Roles) {
+
+    'use strict';
+
+    var permissionUrl = '/admin/api/permissions',
```

Comment:
That should be the easy part .. we just need to generate all the admin routes and encode them in a JSON array. The only "issue" would be if we want to support parameter substitution, but that is also quite easily solvable.

Note there is also the https://github.com/FriendsOfSymfony/FOSJsRoutingBundle which I generally don't like, but it might be a good fit here.